### PR TITLE
[FIX] mrp: confirm MO with deleted WO

### PR DESCRIPTION
**Pull request by paso-odoo**
If there are some dependencies between BoM's operations, a traceback 
will appear when the user removes one WO. 

Steps to produce:
- From Manufacturing > Configuration: Enable Work Orders.
- Open any Bills of Material.
- In 'Miscellaneous' tab tick the 'Operation Dependencies' boolean.
- Set 3 or more operations in the 'Operations' tab.
- After saving the BOM add 'Blocked By' in the operations.
- Now Create Manufacturing Order for that BOM.
- Delete the one or more workorder(s) from mrp order.
- Now click on the 'Confirm' button, and the error will be raised.

The loop is incorrect because we iterate on the BoM's operations but 
the keys of the dict `workorder_per_operation` are based on the MO's 
work orders. Therefore, if we remove one WO, the dict will not have 
any key for the related operation, hence the traceback. 

sentry-4146643254
see - https://tinyurl.com/2ozolrgg

### CODE REVIEW

**Comment by paso-odoo** [comment]
@odoo/sentry_reviewers 

Can you please check this PR?

File: addons/mrp/models/mrp_production.py [hunk 1]
```diff
@@ -1280,7 +1280,8 @@ def _link_workorders_and_moves(self):
         self.allow_workorder_dependencies = self.bom_id.allow_operation_dependencies
         if self.allow_workorder_dependencies:
             for workorder in self.workorder_ids:
-                workorder.blocked_by_workorder_ids = [Command.link(workorder_per_operation[operation_id].id) for operation_id in workorder.operation_id.blocked_by_operation_ids]
+                workorder.blocked_by_workorder_ids = [Command.link(workorder_per_operation[operation_id].id) for operation_id in \
+                                                      workorder.operation_id.blocked_by_operation_ids.filtered(lambda o: o in workorder_per_operation)]
```
Comment (adwid) [inline addons/mrp/models/mrp_production.py:1284]:
```suggestion
                workorder.blocked_by_workorder_ids = [Command.link(workorder_per_operation[operation_id].id) 
                                                      for operation_id in workorder.operation_id.blocked_by_operation_ids
                                                      if operation_id in workorder_per_operation]
```

**Comment by paso-odoo** [comment]
@adwid 

The requested changes are done. 
If you have any suggestions, please let me know.

Thank you!

**Comment by paso-odoo** [comment]
@adwid 

The requested changes are done.
Can you please check this PR for test: https://github.com/odoo/enterprise/pull/41956

**Review by adwid — COMMENTED** [review]
Small change, as the title could be better worded. Also, it is 
always nice to give a little explanaiton about the error (why does 
it occur). Finally, we avoid explanaing what does the commit do (we 
avoid details like "Fix that issue by checking the workcenters which 
is used in the mrp order") because the diff already gives that 
information

Can you update the commit description with this:
```
[FIX] mrp: confirm WO with deleted WO

If there are some dependencies between BoM's operations, a traceback 
will appear when the user removes one WO. 

Steps to produce:
- From Manufacturing > Configuration: Enable Work Orders.
- Open any Bills of Material.
- In 'Miscellaneous' tab tick the 'Operation Dependencies' boolean.
- Set 3 or more operations in the 'Operations' tab.
- After saving the BOM add 'Blocked By' in the operations.
- Now Create Manufacturing Order for that BOM.
- Delete the one or more workorder(s) from mrp order.
- Now click on the 'Confirm' button, and the error will be raised.

The loop is incorrect because we iterate on the BoM's operations but 
the keys of the dict `workorder_per_operation` are based on the MO's 
work orders. Therefore, if we remove one WO, the dict will not have 
any key for the related operation, hence the traceback. 

sentry-4146643254
``` 

You will find all details and guidelines about commit redaction here:
https://www.odoo.com/documentation/16.0/contributing/development/git_guidelines.html

**Comment by adwid** [comment]
I made a little type in my suggestion, sorry
```diff
- mrp: confirm WO with deleted WO
+ mrp: confirm MO with deleted WO
```
Can you rectify ? Then I will r+ both side

**Comment by paso-odoo** [comment]
@adwid 

Changes are done
Thanks!

**Comment by adwid** [comment]
thx
robodoo r+

**Pull request closed without merging**